### DISCUSSION TITLE
Could SHIMLESS-RMA be useful in any way?

### DISCUSSION THREAD

**Discussion by s0urce-c0de** [3kh0/ext-remover#1099] [Help JOIN TITANIUM NETWORK TO SHARE PLEASE] [answered]
I figured out how to boot into chrome://shimless-rma/. Steps:
* Hold power
* Triple tap reload
* Release power
* Wait for the chrome os logo to pop up.
* Wait until the long black screen goes away.
* You should see Chromebook repair.

I found [this](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22This%20repair%20process%20is%20intended%20for%20a%20trained%20repair%20technician.%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F) by plugging in some of the text strings. It lead me to [this](https://source.chromium.org/chromium/chromium/src/+/main:ash/webui/shimless_rma/shimless_rma.cc;l=92?q=IDS_SHIMLESS_RMA_AUTHORIZED_TECH_ONLY_WARNING&sq=&ss=chromium%2Fchromium%2Fsrc) after plugging in a string from this file. Then it finally led me [here](https://source.chromium.org/chromium/chromium/src/+/main:ash/webui/shimless_rma/resources/onboarding_landing_page.html;l=33?q=beginRmaWarningText&ss=chromium%2Fchromium%2Fsrc:ash%2Fwebui%2Fshimless_rma%2F) which I can assume is the source code for shimless rma. I started from [here](https://support.google.com/chromebook/answer/3227606), but I mispressed the shortcut. Please respond to this if you find anything similar that is useful! I found this a few weeks ago, but I never thought about this until now.

**EDIT:**

To enter quick settings increase/decrease volume and click the arrow and click back to get to quick settings.

Ideas:
* [Unlock RSU to disable write protection](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_RSU_WP_DISABLE_METHOD_OPTION%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F)
* [Install custom firmware](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_FIRMWARE_UPDATE_INSTALL_IMAGE_TITLE%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F)

**Comment by AshtonDavies** [comment]
~~Why does no one seem to have noticed this repair menu before? Maybe shimless RMA is useful for people who cannot get a flashed USB for exploits.~~

Edit: nah it's useless but you guys have one thing right: this is old and outdated and @s0urce-c0de is done skidding pretty sure

**Reply by s0urce-c0de** [reply]
it is well hidden. everyone sees chrome://shimless-rma/

**Reply by s0urce-c0de** [reply]
i found it by mistake. also, i found a reddit post about this but i dont have a link. people thought it was sh1mmer failed.

**Reply by s0urce-c0de** [reply]
i only have a sd card that always says "the device you inserted does not contain chromeos" whenever i flash a raw/kvs/sh1mmered shim on id. :(

**Reply by AshtonDavies** [reply]
> i only have a sd card that always says "the device you inserted does not contain chromeos" whenever i flash a raw/kvs/sh1mmered shim on id. :(

You're supposed to turn off OS verification, then return to the beginning, and then insert the SD card.

**Reply by s0urce-c0de** [reply]
how do i do that i don't know how to "glitch back to the beginning"

**Reply by AshtonDavies** [reply]
> how do i do that i don't know how to "glitch back to the beginning"

Esc + Refresh + Power again to get to the original recovery screen with OS verification off.

**Reply by s0urce-c0de** [reply]
oh, so turn it off, original recovery screen, then sh1mmer?

**Reply by AshtonDavies** [reply]
> oh, so turn it off, original recovery screen, then sh1mmer?

Yes

**Reply by S-PScripts** [reply]
> Why does no one seem to have noticed this repair menu before? Maybe shimless RMA is useful for people who cannot get a flashed USB for exploits.

what's funny was  yesterday after i saw brandon on his repo exploits and hacks adding add-supervision (a webview which didn't work for me), i went to chromeabout and checked every single chrome URL. i found that shimlessrma opened an app but had err_invalid_url

not really related but eh

also i found another chrome URL that does an insta oh snap but there was something else that loaded for one frame before the error, not sure what it was

**Reply by s0urce-c0de** [reply]
@S-PScripts please read the instructions. 

Also, can someone help find how to [Unlock RSU to disable write protection](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_RSU_WP_DISABLE_METHOD_OPTION%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F) or [Install custom firmware](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_FIRMWARE_UPDATE_INSTALL_IMAGE_TITLE%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F)?

**Reply by S-PScripts** [reply]
i read the instructions after commenting that
probably won't do it though

**Reply by S-PScripts** [reply]
> @S-PScripts please read the instructions.
> 
> Also, can someone help find how to [Unlock RSU to disable write protection](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_RSU_WP_DISABLE_METHOD_OPTION%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F) or [Install custom firmware](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_FIRMWARE_UPDATE_INSTALL_IMAGE_TITLE%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F)?

RSU stands RMA Server Unlock so you're saying how to "Unlock RMA Server Unlock"? Do you mean how to "DO RSU
maybe check this https://chromium.googlesource.com/chromiumos/platform/factory/+/HEAD/setup/RMA_SHIM.md

**Comment by Blobby-Boi** [comment]
It just opens the normal recovery screen

**Reply by s0urce-c0de** [reply]
You did it wrong, You have to be holding the power button while triple clicking the refresh button, then release  it.

**Reply by Blobby-Boi** [reply]
I got it to work but what can you do with it

**Reply by s0urce-c0de** [reply]
idk, maybe do sh1mmer without usb?
[Unlock RSU to disable write protection](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_RSU_WP_DISABLE_METHOD_OPTION%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F)?
[Install custom firmware](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/chromeos_strings.grd?q=%22IDS_SHIMLESS_RMA_FIRMWARE_UPDATE_INSTALL_IMAGE_TITLE%22&ss=chromium%2Fchromium%2Fsrc:chromeos%2F)?

**Comment by bobthe123** [comment]
I also found [this](https://documents.cdn.ifixit.com/HPkPqK4N1PXyufRy.pdf)

**Reply by s0urce-c0de** [reply]
Thanks! How did you find it?

**Reply by bobthe123** [reply]
Google dorking basically, after I was looking for hidden chrome:// URLs and was looking for info about Shimless RMA

**Reply by S-PScripts** [reply]
searching "shimless chromeos guide" gives you the first result as an ifixit pdf (https://documents.cdn.ifixit.com/HPkPqK4N1PXyufRy.pdf)
this pdf is the one bobthe123 sent lolll

**Reply by s0urce-c0de** [reply]
@bobthe123 what was the search query? I can only find it with `site:ifixit.com` or adding `workflow` to my query.

And also:
https://chromium.googlesource.com/chromium/src/+/7c0a79cc7c5427fbf38dbbb18ef07d1f09a024b3/chrome/common/url_constants.cc
https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/webui_url_constants.cc;l=52?q=chrome:%2F%2Fchrome-signin%20chrome:%2F%2Fpolicy&sq=

**Reply by bobthe123** [reply]
Searching `"shimless" rma`, on the second page going to the iFixit 'Chromebook Repair' and scrolling down shows this:
<img width="250" alt="image" src="https://github.com/3kh0/ext-remover/assets/172151202/48dfedd9-1a0a-452e-8f44-201ed6086b2c">

**Reply by S-PScripts** [reply]
> @bobthe123 what was the search query? I can only find it with `site:ifixit.com` or adding `workflow` to my query.
> 
> And also: https://chromium.googlesource.com/chromium/src/+/7c0a79cc7c5427fbf38dbbb18ef07d1f09a024b3/chrome/common/url_constants.cc https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/webui_url_constants.cc;l=52?q=chrome:%2F%2Fchrome-signin%20chrome:%2F%2Fpolicy&sq=

dude r u blind i literally told you the prompt

**Reply by S-PScripts** [reply]
![image](https://github.com/3kh0/ext-remover/assets/136199723/91ef4684-d6ba-41c6-a978-6fea3e199909)
did you seriously not see this???

**Reply by s0urce-c0de** [reply]
@S-PScripts 
![](https://github.com/3kh0/ext-remover/assets/122696391/7eb9e8b2-a434-48b2-9b91-c4bbb2790c1b)

**Reply by S-PScripts** [reply]
oh ok

**Comment by nathblade16** [comment, marked as answer]
alternative method: buy your own laptop and use it for school

**Comment by appleflyerv3** [comment]
no shimless rma isnt useful stop skidding thanks

**Reply by appleflyerv3** [reply]
unless you find a bug in shimless rma that allows you to do shit

**Reply by S-PScripts** [reply]
> no you can be quiet. Its not skidding it is a question

jeez don't i love it when someone decides to randomly reply almost TWO MONTHS after a comment

**Reply by s0urce-c0de** [reply]
@inLnx Please no necroposting.

**Reply by appleflyerv3** [reply]
> no you can be quiet. Its not skidding it is a question

erm axcually you are skidding 🤓

my answer was marked as the question answer so its true now keep quiet 😀🤞🤩
